src/components/DatePicker.js

```diff
@@ -1206,37 +1211,36 @@ export default class DatePicker extends Component {
 
     if (isDisabledDay) {
       return (
-        <span
+        <DisabledDay
           key={ `day-${index}` }
-          ref={ dateKey }
           style={ dayStyle }
-          onMouseEnter={ (event) => this._onDayMouseEnter(dateKey, event) }
-          onMouseLeave={ (event) => this._onDayMouseLeave(dateKey, event) }
-          {...this.disabledDayProps}
+          dateKey={ dateKey }
+          onDayMouseEnter={ this._onDayMouseEnter }
+          onDayMouseLeave={ this._onDayMouseLeave }
+          disabledDayProps={ this.disabledDayProps }
         >
           { renderedDay }
-        </span>
+        </DisabledDay>
       );
     }
 
```

Comment:
I'm open to a PR to simplify the whole structure. I'm not a big fan of having these components just for event handling, but it's the best option I'm aware of to avoid performance issue and therefor more important than not doing it at all.